services/fluxmonitor_test.go

```diff
@@ -164,62 +168,233 @@ func TestNewFluxMonitor(t *testing.T) {
+		fmConfig.Threshold = tt.threshold
+		fmConfig.AbsoluteThreshold = tt.absoluteThreshold
+		fmConfig.Heartbeat = tt.heartbeat
+		fmConfig.PollInterval = tt.heartbeat
+		fmConfig.RuntimeConfig = store.RuntimeConfig{FMAdapterTimeout: 100 * time.Millisecond, FMAdapterRetryAttempts: 1, FMAdapterRetryDelay: 10 * time.Millisecond}
 
+		t.Run("1 new round event tests: "+tt.name, func(t *testing.T) {
+			fmt.Printf("Testing %s", t.Name())
 			fm, err := NewFluxMonitor("test", fmConfig, triggerJobRun, &mockBlockchainManager{})
 			require.NoError(t, err)
 			defer fm.Stop()
 			wg := sync.WaitGroup{}
 			wg.Add(1)
 			fmt.Println(prettyPrint(fm.state))
-			fmt.Println("New round event, initiated: ", fm.state.RoundID+1)
+			fmt.Println("New round event, not initiated: ", fm.state.RoundID+1)
 			FAEvents <- common.FMEventNewRound{
 				RoundID:         fm.state.RoundID + 1,
 				OracleInitiated: false,
 			}
-
 			go func() {
 				defer wg.Done()
-				timeout := time.NewTimer(2 * time.Second)
-				defer timeout.Stop()
-
-				select {
-				case job := <-triggerJobRun:
-					fmt.Println("Job triggered", job["result"])
-					if tt.want == "no_job" {
-						t.Errorf("Job received. Want %s", tt.want)
-					}
-					if got := job["result"]; !reflect.DeepEqual(fmt.Sprintf("%s", got), fmt.Sprintf("%s", tt.want)) {
-						t.Errorf("GetTriggerJson() = %s, want %s", got, tt.want)
-					}
-
-				case <-timeout.C:
-					fmt.Println("Job timeout")
-					if tt.want != "no_job" {
-						t.Errorf("No Job received, timeout. Want %s", tt.want)
-					}
-				}
+				waitForTrigger(t, triggerJobRun, tt.want, COMMON_JOB_TRIGGER_TIMEOUT)
+			}()
+			wg.Wait()
+		})
 
+		t.Run("2 rounds tests: "+tt.name, func(t *testing.T) {
+			fmt.Printf("Testing %s", t.Name())
+			fm, err := NewFluxMonitor("test", fmConfig, triggerJobRun, &mockBlockchainManager{})
+			require.NoError(t, err)
+			defer fm.Stop()
+			wg := sync.WaitGroup{}
+			wg.Add(1)
+			fmt.Println("FluxMonitor state: ", prettyPrint(fm.state))
+			fmt.Println("1st round event, non initiated: ", fm.state.RoundID+1)
+			FAEvents <- common.FMEventNewRound{
+				RoundID:         fm.state.RoundID + 1,
+				OracleInitiated: false,
+			}
+			go func() {
+				defer wg.Done()
+				waitForTrigger(t, triggerJobRun, tt.want, COMMON_JOB_TRIGGER_TIMEOUT)
+			}()
+			wg.Wait()
+			wg.Add(1)
+			fmt.Println("FluxMonitor state: ", prettyPrint(fm.state))
+			fmt.Println("2nd round event, non initiated: ", fm.state.RoundID+1)
+			FAEvents <- common.FMEventNewRound{
+				RoundID:         fm.state.RoundID + 2,
+				OracleInitiated: false,
+			}
+			go func() {
+				defer wg.Done()
+				waitForTrigger(t, triggerJobRun, tt.want, COMMON_JOB_TRIGGER_TIMEOUT)
 			}()
 			wg.Wait()
 		})
+
+		// TODO: could be handled better?
+		// we want to check if job is triggered after only certain event, therefore makes sense to test only cases that do not have ticker triggers
+		if strings.Contains(tt.name, "no heartbeat, no polling") {
```

Comment:
Instead of checking the name of the test, we should rather have a bool flag in the test.

Also, in Go, we want to keep the happy path the left path. So instead of indenting the rest of the tests in a new block, we would simply check for the opposite and continue on to the next set of tests if we shouldn't run this extra set of tests. Although the case here isn't exactly "happy path", it works pretty well as a best practice with Go code in general.

And it seems like a lot of the feedback above applies to the tests below as well.